runbotics/runbotics-actions-windows/src/automation/office/excel.action-handler.ts

```diff
@@ -116,15 +115,66 @@ export default class ExcelActionHandler extends StatefulActionHandler {
         this.switchPrevWorksheet();
     }
 
+    async createWorksheet(
+        input: ExcelCreateWorksheetActionInput
+    ): Promise<ExcelCreateWorksheetActionOutput> {
+        const worksheets = this.session.Worksheets;
+        const worksheetsCount = worksheets.Count;
+        let worksheet: string;
+
+        if (!input.activate)
+            this.setPrevWorksheet(this.session.ActiveSheet.Name);
+
+        if (input.name) {
+            if (this.checkIfWorksheetExist(input.name))
+                throw new Error("Worksheet name already exists.");
+
+            worksheet = this.session.Worksheets.Add(
+                null,
+                this.session.Worksheets(worksheetsCount)
+            ).Name = input.name;
+        } else {
+            worksheet = this.session.Worksheets.Add(
+                null,
+                this.session.Worksheets(worksheetsCount)
+            ).Name;
+        }
+
+        if (!input.activate) this.switchPrevWorksheet();
+
+        return worksheet;
+    }
+
+    async renameWorksheet(
+        input: ExcelRenameWorksheetActionInput
+    ): Promise<void> {
+        if (input.worksheet && !this.checkIfWorksheetExist(input.worksheet))
+            throw new Error("Worksheet doesn't exist.");
+
+        if (
+            input.newName.trim() === "" ||
+            this.checkIfWorksheetExist(input.newName)
+        )
+            throw new Error("Worksheet name incorrect or already taken.");
```

Comment:
same as few comments above

i'd be great to have function that verifies this conditions and throw error if necessary